<a href="https://colab.research.google.com/github/LeafarxC/aula_dados_analytics_python/blob/main/instalacao_importacao_numpy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#Importação do Numpy
import numpy as np

In [4]:
#Diferença no tempo de execução entre uma lista e um array numpy
from time import time
inicio = time()
lista1 = list(range(10**8))
soma1 = 0

for i in lista1:
    soma1 += i

fim = time()
tempo1 = fim - inicio
print(f'Tempo de execução da lista: {tempo1}')


Tempo de execução da lista: 16.685920238494873


In [5]:
#Diferença no tempo de execução entre uma lista e um array numpy
inicio2 = time()
arr = np.array(lista1, dtype=float)
soma2 = arr.sum()

fim2 = time()
tempo2 = fim2 - inicio2
print(tempo2)

7.855189561843872


In [6]:
print(tempo2 / tempo1)

0.47076753631614127


arra.numpy executa 47% ( no codigo acima) mais rápido